packages/reakit/src/_utils/fastFilter.js

```diff
@@ -0,0 +1,26 @@
+const fastFilter = (func, list) => {
+  let index = 0;
+  let tempIndex = 0;
+  const { length } = list;
+  const temp = new Array(length);
+
+  while (index < length) {
+    if (func.call(null, list[index], index, list)) {
+      temp[tempIndex] = list[index];
+      tempIndex += 1;
+    }
+    index += 1;
+  }
+
+  // Remove empty elements at the end of temp. Like a fast slice.
+  index = 0;
+  const result = new Array(tempIndex);
+
+  while (index < tempIndex) {
+    result[index] = temp[index];
+    index += 1;
+  }
+  return result;
+};
+
+module.exports = fastFilter;
```

Comment:
Let's use `default export`.

Reply:
If you're importing it with commonjs (`require`), you'll need to add `.default` to the required object.